### PR TITLE
Add merging and saving to dataservice

### PR DESCRIPTION
**Pull request by jannessm**

### CODE REVIEW

File: src/app/services/file-synchronizer.service.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    });
+  }
+
+  public deleteSonggroup(songgroupid: string) {
+    return this.dexieService.delete(DATABASES.songgroups, songgroupid).then(() => {
+      this.apiService.generateDeleteFileRequest(path.join(DATABASES.songgroups, songgroupid + '.songgroup'));
+    });
+  }
+
+  private map<T>(array) {
```
Comment (nilsroesel) [inline src/app/services/file-synchronizer.service.ts]: you can shorten this if you want
```suggestion
  private map<T>(array): Array<T> { return (array || []).map(v => <T>v) }
```

Comment (nilsroesel) [reply src/app/services/file-synchronizer.service.ts]: maybe the naming should also be more like 'castMap' or something similar

File: src/app/services/parser.service.ts [hunk 2]
```diff
@@ -180,29 +180,32 @@ export class ParserService {
...
     // meta
     const title = song.title && song.title !== '' ? 'title: ' + song.title + '; ' : '';
     const artist = song.artist && song.artist !== '' ? 'artist: ' + song.artist + '; ' : '';
-    const bpm = song.bpm && song.bpm ? 'bpm: ' + song.bpm + '; ' : '';
-    const books = song.books && song.books ? 'books: ' + song.books.join(',') + '; ' : '';
-    str += '[' + title + artist + bpm + books + ']\n\n';
+    const bpm = song.bpm ? 'bpm: ' + song.bpm + '; ' : '';
+    const books = song.books && song.books.length > 0 ? 'books: ' + song.books.join(',') + '; ' : '';
+
+    if (title || artist || bpm || books) {
```
Comment (nilsroesel) [inline src/app/services/parser.service.ts]: is it intended that the str could look at the end like
"[ fooundefinedundefinedundefined]"

Comment (jannessm) [reply src/app/services/parser.service.ts]: have you looked at the initialisation of these variables???

Comment (nilsroesel) [reply src/app/services/parser.service.ts]: ah okay, i see :D

**Pull request merged**